pkg/observer/controller.go

```diff
@@ -243,6 +247,63 @@ func (c *controller) dropInProgressNodeGroups(nodeGroups v1.NodeGroupList, cnrs
 	return restingNodeGroups
 }
 
+// get the cluster-autoscaler last scaleUp activity time
+func stringToTime(s string) (time.Time, error) {
+	sec, err := strconv.ParseInt(s, 10, 64)
+	if err != nil {
+		return time.Time{}, err
+	}
+	return time.Unix(sec, 0), nil
+}
+
+// query cluster-autoscaler metrics to figure out if it's safe to start a new CNR
+func (c *controller) safeToStartCycle() bool {
+	safeToStart := true
+	client, err := api.NewClient(api.Config{
+		Address: c.PrometheusAddress,
+	})
+	if err != nil {
+		// Prometheus might not be installed in the cluster. return true if it can't connect
+		klog.Errorln("Error creating client:", err)
+		return safeToStart
+	}
+
+	v1api := promv1.NewAPI(client)
```

Comment:
If you give the Controller struct a promAPI field, you can construct the client/api and do all this checking once at startup (in the main thread), instead of doing it each loop.